src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -26,20 +26,60 @@
 //------------------------------------------------------------------------------
 
 using System;
-using System.Collections.Concurrent;
 using System.Globalization;
 using Microsoft.IdentityModel.Logging;
+#if NET461 || NET472 || NETSTANDARD2_0 
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+#endif
 
 namespace Microsoft.IdentityModel.Tokens
 {
     /// <summary>
     /// Defines a cache for crypto providers.
     /// Current support is limited to <see cref="SignatureProvider"/> only.
     /// </summary>
-    public class InMemoryCryptoProviderCache : CryptoProviderCache
+#if NET461 || NET472 || NETSTANDARD2_0 
+    public class InMemoryCryptoProviderCache : CryptoProviderCache, IDisposable
+#elif NET45
+    public class InMemoryCryptoProviderCache: CryptoProviderCache
```

Comment:
Are SignatureProviders stored in MaxCapacityRefreshCache going to be properly disposed in net45?

Reply:
I think so? The InMemoryCryptoProviderCache is only responsible for disposing the cache itself, not the SignatureProviders contained inside of it. I believe each SignatureProvider is responsible for disposing itself, so this behavior should remain consistent across all targets.

Reply:
I added two tests that show SignatureProviders are disposed of properly across all targets. These are currently present as unit tests to make review easier.

They make use of GC.Collect() and GC.WaitForPendingFinalizers(), so both tests should definitely be moved to a different location later to avoid slowing down our tests.